### PR TITLE
Change install to one env file & update docs

### LINKED ISSUE #481: RTD: Change "Create Virtual Environment and ..." to Install AxonDeepSeg

### LINKED ISSUE #482: RTD: Remove conda activate from installation section, and make a new section for using ADS

### LINKED ISSUE #483: Installation with OS-specifc environment files fails
Following the merge of PR #452, the installation with OS-specifc environment yml files failed with many conflicts on Ubuntu 20.04. On MacOS Catalina 10.15.7, the installation worked with a warning, but the axondeepseg_test failed. Details below.

[ads-install-log_ubuntu-20.04.txt](https://github.com/neuropoly/axondeepseg/files/6184280/ads-install-log_ubuntu-20.04.txt)

<details>
  <summary>Catalina warning</summary>

```
(base) λ ~/axondeepseg/ master conda env create -f environment_macOS.yml -n ads_venv
Collecting package metadata (repodata.json): - WARNING conda.models.version:get_matcher(537): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 2.*, but conda is ignoring the .* and treating it as 2
- WARNING conda.models.version:get_matcher(537): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.*, but conda is ignoring the .* and treating it as 1
/ WARNING conda.models.version:get_matcher(537): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 3.6.*, but conda is ignoring the .* and treating it as 3.6
done
Solving environment: done
```
</details>

[catalina_axondeepseg_test.txt](https://github.com/neuropoly/axondeepseg/files/6184282/catalina_axondeepseg_test.txt)

TODO:

- [ ] Remove OS-specific environment files
- [ ] Update documentation to use `environment.yml` to install on all OS
- [ ] Add a warning in doc for the potential long installation time

### PR DESCRIPTION
**Pull request by mathieuboudreau**
## Checklist

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [ ] I've added relevant tests for my contribution
- [x] I've updated the documentation and/or added correct docstrings
- [ ] I've assigned a reviewer
- [x] I've consulted [ADS's internal developer documentation](https://github.com/neuropoly/axondeepseg/wiki/Guidelines-for-contribution) to ensure my contribution is in line with any relevant design decisions

Resolves #481, Resolves #482, Resolves #483

### CODE REVIEW

**Comment by mathieuboudreau** [comment]
@jcohenadad I had to leave activating the conda env in the installation step, as the installation step needs another command within in (`pip install -e .`). However, I've added a small section under the "Using AxonDeepSeg" section to say that you need to activate the venv if not already activated.

**Comment by mariehbourget** [comment]
We made the changes to use only the common `environment.yml` file to install ADS on all OS.
I tested it on my laptop in Ubuntu 20.04 in WSL1 and Windows 10, everything is working as expected.

@vs74, @Stoyan-I-A, @dyt811, @ahill187, @joshuacwnewton, @gisellemartel Could you re-test the installation using this branch? The rendered doc is [here](https://axondeepseg--484.org.readthedocs.build/en/484/documentation.html#installation). Thank you everyone for your help!

**Comment by vs74** [comment]
Installed ADS by following the docs. I had no issues with the installation on **MacOS Mojave**. Tested `axondeepseg_test` and `flseyes` too; everything works like a charm.

**Comment by joshuacwnewton** [comment]
Installing worked for me on Ubuntu 20.04, as well. :slightly_smiling_face:

**Comment by Stoyan-I-A** [comment]
The installation worked. Tested on Windows 10.

**Comment by gisellemartel** [comment]
Installation worked for me on MacOS Catalina 10.15.7!

**Comment by Stoyan-I-A** [comment]
> Installation worked for me on MacOS Catalina 10.15.7! Testing the installation however failed for `axondeepseg_test`. Here is a log: [log.txt](https://github.com/neuropoly/axondeepseg/files/6186380/log.txt)

Have you used the Theano backend in the past? It looks like the software is trying to use Theano instead of tensorflow.

**Comment by mathieuboudreau** [comment]
> > Installation worked for me on MacOS Catalina 10.15.7! Testing the installation however failed for `axondeepseg_test`. Here is a log: [log.txt](https://github.com/neuropoly/axondeepseg/files/6186380/log.txt)
> 
> Have you used the Theano backend in the past? It looks like the software is trying to use Theano instead of tensorflow.

Hmm interesting, could this be a tool that was installed with `pip` and the `--user` install? I've encountered instances where those have leaked into my venvs, very annoying and tough to troubleshoot if you're not aware/remember.

**Comment by mariehbourget** [comment]
I found [this](https://stackoverflow.com/questions/43327464/how-to-make-keras-use-tensorflow-backend-in-anaconda), it seems to be an issue when installing Keras through conda-forge, though the post is 3 years old.

The only way I found to switch the backend permanently was to explicitly export the variable in `${CONDA_PREFIX}/etc/conda/activate.d/keras_activate.sh` with `export KERAS_BACKEND="tensorflow"`

**Comment by mathieuboudreau** [comment]
> I found [this](https://stackoverflow.com/questions/43327464/how-to-make-keras-use-tensorflow-backend-in-anaconda), it seems to be an issue when installing Keras through conda-forge, though the post is 3 years old.
> 
> The only way I found to switch the backend permanently was to explicitly export the variable in `${CONDA_PREFIX}/etc/conda/activate.d/keras_activate.sh` with `export KERAS_BACKEND="tensorflow"`

@gisellemartel could you try this fix temporarily to see if resolves the issue? If so, @mariehbourget let's just make a quick Notice for it on the ReadTheDocs, as it seems like a rare issue and we're migrating away from Keras when we move to IVADOMED soon.

**Comment by mariehbourget** [comment]
@gisellemartel, did you get the chance to test the fix describe [above](https://github.com/neuropoly/axondeepseg/pull/484#issuecomment-804955746)?

**Comment by gisellemartel** [comment]
> @gisellemartel, did you get the chance to test the fix describe [above](https://github.com/neuropoly/axondeepseg/pull/484#issuecomment-804955746)?

Thank you @mariehbourget @mathieuboudreau @Stoyan-I-A for your suggestions. I tried the suggested fix and it worked! I do still get some warnings after running the test however
[log.txt](https://github.com/neuropoly/axondeepseg/files/6198901/log.txt)

**Comment by mariehbourget** [comment]
Thanks @gisellemartel for the tests.

@mathieuboudreau, I added the warning to the doc. Can you review it and let me know if that is ok with you (I can't assign you officially since you opened the PR). Thanks!

File: docs/source/documentation.rst [hunk 1]
```diff
@@ -115,6 +115,8 @@ This integrity test automatically performs the axon and myelin segmentation of a
 
     * * * Integrity test passed. AxonDeepSeg is correctly installed. * * * 
 
+.. NOTE :: For some users, the test may fail because Keras is using Theano backend instead of Tensorflow. In that case, you will see the line ``Using Theano backend.`` when launching ``axondeepseg_test``. To fix this issue, add the line ``export KERAS_BACKEND="tensorflow"`` at the end of the ``<your_conda_install_location>\envs\<your_environment_name>/etc/conda/activate.d/keras_activate.sh`` file, then deactivate and reactivate your environment. The test should print ``Using Tensorflow backend.`` and pass.
+
```
Comment (joshuacwnewton) [inline docs/source/documentation.rst:119]: Do we need to warn about this?

IIRC, the Keras project has moved on from non-Tensorflow backends ever since it was integrated into TF. 

Though, now that I'm looking into this, they recently edited their README to [remove information about this](https://github.com/keras-team/keras/commit/4fd825ddf93d3dc9ecc7b50178ce1d015393bed2#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5). They used to have this warning:

> **Multi-backend Keras has been discontinued. At this time, we recommend that Keras users who use multi-backend Keras with the TensorFlow backend switch to `tf.keras` in TensorFlow 2.0**.

But now it's ["under construction"](https://github.com/keras-team/keras/blob/master/README.md). Huh.

Comment (mathieuboudreau) [reply docs/source/documentation.rst:119]: We decided to warn about this because in particular, one of our test users encountered exactly this issue here in this thread =)

**Comment by mathieuboudreau** [comment]
@mariehbourget The message reads well for me; if the tests pass you can approve and merge, which will open the door to merging #479.

File: docs/source/documentation.rst (hunk 1 above)
Comment (joshuacwnewton) [reply docs/source/documentation.rst:119]: I guess what I'm saying is, could this be an isolated incident that doesn't apply to most users? :stuck_out_tongue:

**Comment by mathieuboudreau** [comment]
Haha fair. Maybe it coyld be in a troubleshooting section, but lets merge it here for now as the other PR is urgent

Téléchargez Outlook pour Android<https://aka.ms/ghei36>
________________________________

**Review by mariehbourget — APPROVED** [review]
